cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -68,20 +69,21 @@ func CreateAdminAndKubeletKubeConfig(masterEndpoint, pkiDir, outDir string) erro
 	}
 
 	// User admin should have full access to the cluster
-	if err := createKubeConfigFileForClient(masterEndpoint, "admin", outDir, caCert, caKey); err != nil {
-		return fmt.Errorf("couldn't create a kubeconfig file for admin: %v", err)
+	if err := createKubeConfigFileForClient(masterEndpoint, AdminKubeConfigFileName, []string{"system:masters"}, outDir, caCert, caKey); err != nil {
+		return fmt.Errorf("couldn't create %s: %v", AdminKubeConfigFileName, err)
 	}
 
-	// TODO: The kubelet should have limited access to the cluster
-	if err := createKubeConfigFileForClient(masterEndpoint, "kubelet", outDir, caCert, caKey); err != nil {
-		return fmt.Errorf("couldn't create a kubeconfig file for kubelet: %v", err)
+	// The kubelet should have limited access to the cluster
+	if err := createKubeConfigFileForClient(masterEndpoint, KubeletKubeConfigFileName, []string{"system:nodes"}, outDir, caCert, caKey); err != nil {
```

Comment:
Yes, it makes sense to separate those two. `admin.conf` and `kubelet.conf` should remain, but the client names inside of the file should be dynamic